library/network/src/main/java/com/epmedu/animeal/network/NetworkStateProviderImpl.kt

```diff
@@ -0,0 +1,26 @@
+package com.epmedu.animeal.network
+
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+class NetworkStateProviderImpl(
+    private val connectivityManager: ConnectivityManager
+) : NetworkStateProvider {
+    override fun getNetworkState(): Flow<NetworkState> {
+        return flow {
```

Comment:
Why flow? Is there a stream of data?

Reply:
I considered other cases through the feeding flow where it's possible network is not available and we'll need to fix later. Even it's possible to make it static with a single line, but this is more flexible.